torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -194,10 +195,18 @@ bool needsProfiledOutput(Node* n) {
 void ProfilingRecord::instrumentBlock(Block* block) {
   for (auto it = block->nodes().begin(); it != block->nodes().end(); ++it) {
     auto n = *it;
-    for (auto i : n->inputs()) {
+    for (size_t offset = 0; offset < n->inputs().size(); offset++) {
+      auto i = n->input(offset);
+      // n can have multiple uses, if we already
+      // processed `n` we also inserted `prim::profile`s
+      // let's check for this case otherwise we will
+      // be `prim::profile` `prim::profile`
+      if (i->node()->kind() == prim::profile) {
```

Comment:
Does it ever fire? I suspect that it shouldn't once we changed the way we replace inputs (previously we were replacing all inputs of a given value, now we're only replacing one specific input with a given index).

Reply:
yeah, giving this some more thought, it **shouldn't** fire. I also double checked by adding an assertion in that if and it never fired (modulo windows failures) which don't seem to be related to my changes.